src/dune_rules/melange/melange_rules.ml

```diff
@@ -247,6 +247,59 @@ let setup_emit_cmj_rules ~sctx ~dir ~scope ~expander ~dir_contents
   in
   Buildable_rules.with_lib_deps ctx compile_info ~dir ~f
 
+module Runtime_deps = struct
+  let to_action_builder ~expander dep_conf =
+    let runtime_deps, _sandbox =
+      Dep_conf_eval.unnamed_get_paths ~expander dep_conf
+    in
+    runtime_deps
+
+  let eval ~expander (deps : Dep_conf.t list) =
+    let open Memo.O in
+    let builder = to_action_builder ~expander deps in
+    let+ paths, _ = Action_builder.run builder Eager in
+    paths
+
+  let targets ~output deps =
+    Path.Set.to_list deps
+    |> List.filter_map ~f:(fun src ->
+           match output with
+           | `Private_library_or_emit output_dir -> (
+             match Path.as_in_build_dir src with
+             | Some src_build ->
+               let target = Path.Build.drop_build_context_exn src_build in
+               Some (src, Path.Build.append_source output_dir target)
+             | None -> None)
```

Comment:
Instead of dropping, you should add them to as deps to the alias directly without the copying rules

Reply:
ah that makes sense

Reply:
fixed this one in [76ef993](https://github.com/ocaml/dune/pull/7199/commits/76ef993a276e3abb54bf1a22408ff113f34bb6cd)